### PR TITLE
feat: accept only string or undefined as parameter of stack

### PR DESCRIPTION
**Pull request by polRk**
interface Error {
    name: string;
    message: string;
    stack?: string;
}

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why?

**Comment by polRk** [comment]
> Why?

Because your typings are not compatible. See https://github.com/microsoft/TypeScript/blob/master/lib/lib.es5.d.ts#L972

**Comment by sindresorhus** [comment]
You cannot just change this though. The JS part doesn't handle `undefined` input.

**Comment by polRk** [comment]
Emmm, 
```
cleanStack(input: string) // input only strings

cleanStack(error.stack) // error.stack - can be undefined // Typescript error
```

As the result i can not use your library with typescript

**Comment by sindresorhus** [comment]
In case I was not clear enough. I’m fine with this change, but you need to change the JS side too to handle `undefined`.

**Comment by polRk** [comment]
Hmm, i do not know, is it possible to create preload module like dotenv 
```
node -r clean-stack/register index.js
```

**Comment by sindresorhus** [comment]
The docs needs to be updated too, to mention that if the input is not a string, it will return undefined.

File: index.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ const pathRegex = /^(?:(?:(?:node|(?:internal\/[\w/]*|.*node_modules\/(?:babel-p
 const homeDir = typeof os.homedir === 'undefined' ? '' : os.homedir();
 
 module.exports = (stack, options) => {
+	if (!stack || typeof stack !== 'string') {
```
Comment (sindresorhus) [inline index.js:9]:
```suggestion
	if (typeof stack !== 'string') {
```

**Comment by sindresorhus** [comment]
@polRk Bump :)

**Comment by sindresorhus** [comment]
Closing for lack of response.

https://github.com/sindresorhus/clean-stack/issues/20

**Pull request closed without merging**